### ISSUE TITLE
False positive invalid-name with dataclass ClassVar

### ISSUE THREAD

**Issue by jamesbraza** [PyCQA/pylint#4154]
### Steps to reproduce

Happens when you use `dataclasses.dataclass` and `ClassVar` to represent a constant field.

Given a file `a.py`:

```python
# pylint: disable=missing-module-docstring,missing-class-docstring

from dataclasses import dataclass
from typing import ClassVar

@dataclass
class Foo:
    BAR: ClassVar[int] = 42
```

Came from here: https://github.com/PyCQA/pylint/issues/4149#issuecomment-786952039

Relevant PR: https://github.com/PyCQA/astroid/pull/914

### Current behavior

Result of `pylint a.py`:

```
************* Module a
a.py:8:4: C0103: Attribute name "BAR" doesn't conform to snake_case naming style (invalid-name)
```

### Expected behavior

I expect no `invalid-name` message, as `ClassVar` is allowed within `dataclass`: [`dataclasses` Class variables](https://docs.python.org/3/library/dataclasses.html#class-variables)

This is per [PEP8 Constants](https://www.python.org/dev/peps/pep-0008/#constants).

### pylint --version output

Result of `pylint --version` output:

```
pylint 2.7.1
astroid 2.5
Python 3.8.6 (default, Oct 27 2020, 11:18:42)
[Clang 12.0.0 (clang-1200.0.32.27)]
```
